### PR TITLE
Run3-gem62B Use of ESGetToken in GEM geometry packages

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Use of ESGetToken in GEM geometry packages

#### PR validation:

Use the runTheMatrix test workflows

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by civanch** [comment]
+1

**Comment by srimanob** [comment]
+Upgrade

**Comment by qliphy** [comment]
+1

File: Geometry/GEMGeometryBuilder/plugins/ME0GeometryValidate.cc [hunk 1]
```diff
@@ -102,17 +102,17 @@ class ME0GeometryValidate : public one::EDAnalyzer<> {
...
       tolerance_(iConfig.getUntrackedParameter<int>("tolerance", 6)) {
   fwGeometry_.loadMap(infileName_.c_str());
   outFile_ = TFile::Open(outfileName_.c_str(), "RECREATE");
 }
 
 void ME0GeometryValidate::analyze(const edm::Event& event, const edm::EventSetup& eventSetup) {
-  eventSetup.get<MuonGeometryRecord>().get(me0Geometry_);
-  if (me0Geometry_.isValid()) {
+  me0Geometry_ = &eventSetup.getData(tokGeom_);
+  if (me0Geometry_ != nullptr) {
```
Comment (makortel) [inline Geometry/GEMGeometryBuilder/plugins/ME0GeometryValidate.cc:115]: This is a change in functionality. The `EventSetup::getData()` throws an exception if the product is not available. In order to check the availability of the product, `EventSetup::getHandle()`, that returns `ESHandle`, should be used .

**Pull request merged**